api/v1beta1/types.go

```diff
@@ -768,6 +771,14 @@ type AzureBastion struct {
 	PublicIP PublicIPSpec `json:"publicIP,omitempty"`
 }
 
+// BackendPool describes the backend pool of the load balancer.
+type BackendPool struct {
+	// Name specifies the name of backend pool for the load balancer. If not specified, the default name will
+	// be set, depending on the load balancer role.
+	// +optional
+	Name *string `json:"name,omitempty"`
```

Comment:
is there value in making this a string pointer given that `""` will never be a valid name? If it was just a ` string`  it would avoid having to dereference it in a bunch of places (and it would be more consistent with other `Name` fields in the API)

Reply:
Yes, I agree. I copied it from the original [PR](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2432) where Name is a pointer. I updated it as you asked.